client_plugin/utils/config/config.go

```diff
@@ -70,3 +86,147 @@ func SetDefaults(config *Config) {
 		config.LogLevel = defaultLogLevel
 	}
 }
+
+// LogInit init log with passed logLevel (and get config from configFile if it's present)
+// returns True if using defaults,  False if using config file
+func LogInit(logLevel *string, logFile *string, defaultLogFile string, configFile *string) bool {
```

Comment:
generally passing multiple string params is error prone. IMO it's much better to define a struct with named params and pass &{name:value,...}.